llvm/lib/Target/Sparc/MCTargetDesc/SparcAsmBackend.cpp

```diff
@@ -202,9 +203,9 @@ namespace {
         { "fixup_sparc_tls_le_lox10",   0,  0,  0 },
         { "fixup_sparc_hix22",         10, 22,  0 },
         { "fixup_sparc_lox10",         19, 13,  0 },
-        { "fixup_sparc_gotdata_hix22",  0,  0,  0 },
-        { "fixup_sparc_gotdata_lox10",  0,  0,  0 },
-        { "fixup_sparc_gotdata_op",     0,  0,  0 },
+        { "fixup_sparc_gotdata_op_hix22", 10, 22,  0 },
```

Comment:
For fixup kinds that always map to relocation types (e.g. GOT-generating and TLS relocation types), the offset/length fields can be kept as 0. The only difference is in the -filetype=asm -show-encoding output, which is not really relevant. (We should test relocations using -filetype=obj; the -filetype=asm -show-encoding output is informal.)

Reply:
If you rebase after 7d423c09c0e063eeeaa2f91100eb6f04dc15767c , you can even remove the fixup kinds.

Reply:
Sounds good, I'll try rebasing on that.